src/System.IO/tests/StreamReader/StreamReader.CtorTests.cs

```diff
@@ -47,5 +48,29 @@ public static void CreationFromMemoryStreamWithEncodingTrue()
             Assert.Equal("ABCD", sr2.ReadToEnd());
             sr2.Dispose();
         }
+
+        [Fact]
+        public void StreamReader_WithOptionalArguments()
+        {
+            using (var sr = new StreamReader(new MemoryStream(), leaveOpen: true))
+            {
+                Assert.Equal(Encoding.UTF8, sr.CurrentEncoding);
+            }
+
+            using (var sr = new StreamReader(new MemoryStream(), encoding: null))
+            {
+                Assert.Equal(Encoding.UTF8, sr.CurrentEncoding);
+            }
+
+            using (var sr = new StreamReader(new MemoryStream(), bufferSize: -1))
+            {
+                Assert.Equal(Encoding.UTF8, sr.CurrentEncoding);
+            }
+
+            using (var sr = new StreamReader(new MemoryStream(), detectEncodingFromByteOrderMarks: true))
+            {
+                Assert.Equal(Encoding.UTF8, sr.CurrentEncoding);
+            }
+        }
```

Comment:
I'm not understanding what these tests are validating.  How does asserting the current encoding == UTF8 help to validate the behavior of specifying each of these optional arguments?

Reply:
I presume you're trying to make sure the defaults are what you expect when you don't specify specific arguments. This is valuable to do. You can easily check the encoding as you have. It is also relatively easy to check the `leaveOpen` default by looking at `MemoryStream.CanRead` after you dispose the reader. That will be false if it has been disposed.

Reply:
As far as `detectEncodingFromByteOrderMarks` goes, it looks like we're weak on testing that in general. I don't see that we have any tests that actually check that DetectEncoding() works as it should. I've created an issue for it: #37190